javascript-review_Moss.js

```diff
@@ -0,0 +1,293 @@
+  
+  // Why do we use promises?
+  // 9.
+  const q9 = 'to make javascript run in asychonise way'
+  
+  // What are the two methods at our disposal if we have hold of a promise? What are the names of the functions that are related to these functions (that are called when the promise is fulfilled or fails).
+  // 10.
+  const q10 = '.then and .catch'
+  
+  // What is the DOM?
+  // 11.
+  const q11 =
+    'DOM is like a parent and child directory tree of html element and also refer to how javascript see html page'
+  
+  // Create an object that has three properties. One property is called ‘count’ and will be set to a number, the second will be set to an array, and the third to a function. This function should simply console.log each of the values of the second property on a new line.
+  
+  // 12.
+  
+  const obj2 = {
+    count: 0,
+    array: ['a', 'b', 'c', 'd', 'e', 'f'],
+    toString: function() {
+      console.log([...this.array].join('\n'))
+    }
+  }
+  // const obj3 = {
+  //   count: 0,
+  //   array: ['a', 'b', 'c', 'd', 'e', 'f'],
+  //   toString: function() {
+  //     this.array.forEach(num => {
+  //       console.log(num)
+  //     })
+  //   }
+  // }
+  // obj2.toString()
+  
+  // What is NPM? In your brief explanation refer to package.json.
+  // 13.
+  let q13 =
+    'NPM is node package manager is where all the plugin is uploaded to and since it is all in one place in can be easily download as well as update(changing version) when require'
+  
+  // What are Event Listeners? Gives some examples in your response.
+  
+  // 14.
+  let q14 =
+    'Event listener is providen feature to listen to event that happen in dom such as click,change'
+  
+  // Define a function called numMult that takes two number arguments and a callback. The function will multiply the two numbers and pass the result to the callback as an argument. Now call numMult and in the callback simply console.log out the argument that has been passed through.
+  // 15.
+  const q15 = num => {
+    console.log(num)
+  }
+  const numMult = (a, b, c) => {
+    return c(a * b)
+  }
+  // numMult(2, 2, q15)
+  // a.
+  // Define a function called addNum with that takes two number arguments. In this function simply return the addition of these numbers.
+  const addNum = (a, b) => {
+    return a + b
+  }
+  // b.
+  // Write a second function called numsPlusFunct that takes three arguments, two numbers and a function. Inside numsPlusFunct call the function that is passed as an argument, and pass the two number arguments to this function. numsPlusFunct will return an object where the first property has the value returned from that function call, and the second property is a string.
+  
+  const numsPlusFunct = (a, b, c) => {
+    return { first: c(a, b), second: 'String' }
+  }
+  // c.
+  // You have now made two functions. Call the numsPlusFunct and pass addNum as the appropriate argument.
+  // console.log(numsPlusFunct(1, 2, addNum))
```

Comment:
✅

just be careful with naming your parameters as a, b and c it makes your code hard to read